rust/protocol/src/lib.rs

```diff
@@ -25,7 +25,7 @@
 mod address;
 mod consts;
 mod crypto;
-mod curve;
+pub mod curve;
```

Comment:
This still counts as changing the module structure; it's the public surface of the module that matters, not its implementation. I still think we should stay flat until we discuss it properly and as a whole.